provider/aci.go

```diff
@@ -1639,38 +1640,39 @@ func getProbe(probe *v1.Probe, ports []v1.ContainerPort) (*aci.ContainerProbe, e
 }
 
 func (p *ACIProvider) getAzureFileCSI(volume v1.Volume, namespace string) (*aci.Volume, error) {
-	azureSource := &aci.AzureFileVolume{
-		ReadOnly: *volume.CSI.ReadOnly,
-	}
-	if volume.CSI.NodePublishSecretRef != nil && volume.CSI.NodePublishSecretRef.Name != "" {
-
-		// Set the storage account name and key
-		secret, err := p.resourceManager.GetSecret(volume.CSI.NodePublishSecretRef.Name, namespace)
-		if err != nil {
-			return nil, errors.Wrap(err, fmt.Sprintf("The secret %s for AzureFile CSI driver %s is not found", volume.CSI.NodePublishSecretRef.Name, volume.Name))
-		}
+	azureSource := &aci.AzureFileVolume{}
 
-		if secret == nil {
-			return nil, fmt.Errorf("Getting secret for AzureFile CSI driver %s returned an empty secret", volume.Name)
-		}
+	if volume.CSI.VolumeAttributes != nil && len(volume.CSI.VolumeAttributes) != 0 {
+		secretName := volume.CSI.VolumeAttributes[azureFileSecretName]
```

Comment:
field name in `CSI.VolumeAttributes` is case insensitive, you could refer to following code to handle such issue:

https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/dabcf18dfa32fac089f45c182b63726726bc29b1/pkg/azurefile/azurefile.go#L568-L588